### PR TITLE
Move parameter constructor to a package extension, use name maps

### PR DESCRIPTION
**Pull request by nefrathenrici**
This PR adds a package extension which includes new convenience constructors for SurfaceFluxesParameters and the various UniversalFunctions parameter structs when CLIMAParameters is loaded.

The core changes are in `ext/CreateParametersExt.jl` and `Project.toml`, but I have updated all of the tests and docs to work with the new constructors.

This PR also alters `recover_profile` to not require a `UniversalFunctionType` as an argument, since it can be obtained from `UniversalFunctionParams` within the function. With a followup PR I would like to remove the UniversalFunctionTypes (BusingerType, ChengType, etc).

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -25,7 +24,6 @@ ArrayType = Array
 @info ArrayType
 
 @testset "SurfaceFluxes - Recovery Profiles" begin
-    FloatType = Float32
```
Comment (charleskawczynski) [inline test/runtests.jl:28]: How is this allowed to be deleted?

Comment (nefrathenrici) [reply test/runtests.jl:28]: `FloatType` gets set on line 14, but I can move it back into the @testset

Comment (charleskawczynski) [reply test/runtests.jl:28]: Ah, I see. Removing it is fine.

File: ext/CreateParametersExt.jl [hunk 2]
```diff
@@ -0,0 +1,137 @@
+module CreateParametersExt
```
Comment (akshaysridhar) [inline ext/CreateParametersExt.jl:1]: Thanks @nefrathenrici, are there any limitations to this that we should be aware of? Perhaps we add a link to `https://pkgdocs.julialang.org/v1.9/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)` to help users.

Comment (nefrathenrici) [reply ext/CreateParametersExt.jl:1]: This is fully backwards-compatible for Julia <1.9.
For 1.9+, CLIMAParameters needs to be loaded beforehand to compile the constructors in the extension. Without this step, users won't have access to the constructors. 
I'm happy to add the link and an explanation for creating parameters to the docs.

Comment (akshaysridhar) [reply ext/CreateParametersExt.jl:1]: Noted, thanks @nefrathenrici

**Review by akshaysridhar — APPROVED** [review]

**Pull request merged**